### PR TITLE
#1198 - Add DoiRedirect component to handle redirecting from DOI landing page links

### LINKED ISSUE #1198: Add ability to take URLs from DOI landing pages and show the correct page
**Description**:
The URLs coming from the current DOI landing pages just have the study or investigation ID, so we will need to have some sort of "landing page" that takes this ID, and constructs the correct url and then redirects to this to show the correct landing page which allows a user to download the data.

**Acceptance criteria**:
- [ ] DOI landing pages can link directly to DG

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Note: this will also require a change in the common-doi repo to remove the # from the DOI redirect URL.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Fixes #1198

### CODE REVIEW

File: packages/datagateway-dataview/src/page/doiRedirect.component.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const loading =
+    facilityCycleLoading ||
+    instrumentLoading ||
+    investigationLoading ||
+    (!!investigation && facilityCycleIdle);
+
+  if (investigation && instrument && facilityCycle) {
+    return (
+      <Redirect
+        to={`/browse/instrument/${instrument.id}/facilityCycle/${facilityCycle.id}/investigation/${investigation.id}`}
```
Comment (sam-glendenning) [inline packages/datagateway-dataview/src/page/doiRedirect.component.tsx:65]: Shall we have this redirect to the dataset view like in topcat?

Comment (louise-davies) [reply packages/datagateway-dataview/src/page/doiRedirect.component.tsx:65]: I discussed this with Alejandra and that is what we decided - I'll push that change shortly

File: packages/datagateway-dataview/src/page/doiRedirect.component.test.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  };
+});
+
+jest.mock('react-router-dom', () => {
+  const originalModule = jest.requireActual('react-router-dom');
+  return {
+    __esModule: true,
+    ...originalModule, // use actual for all non-hook parts
+    useParams: () => ({
+      facilityName: 'ISIS',
```
Comment (sam-glendenning) [inline packages/datagateway-dataview/src/page/doiRedirect.component.test.tsx:40]: Better to use LILS here?

**Comment by louise-davies** [comment]
Yeah I did think about how the homepage did it but as you say it's a separate issue/it works and isn't broken so I didn't fancy trying to fiddle with that 😅

**Review by sam-glendenning — APPROVED** [review]
Brilliant fix

**Pull request merged**